src/Symfony/Component/Validator/Constraint.php

```diff
@@ -108,7 +108,7 @@ public function __construct($options = null)
         $defaultOption = $this->getDefaultOption();
         $invalidOptions = [];
         $missingOptions = array_flip((array) $this->getRequiredOptions());
-        $knownOptions = get_object_vars($this);
+        $knownOptions = get_class_vars(static::class);
```

Comment:
it might also work with private properties as they are accessible from this scope, is that supported? maybe add a test to ensure it cannot regress

Reply:
Good point, I didn't think of that.

Reply:
I've just checked locally, actually private properties aren't accessible from this scope, only public properties are since `Constraint` is higher in the inheritance chain.
Thx for the quick fix :)

Reply:
thanks for checking 👍